File: spark-java/spark-dataframe/dataframe-aggregation/src/main/java/com/mahfooz/df/aggregation/config/RetainColumnGroupEx.java
package com.mahfooz.df.aggregation.config;

import org.apache.spark.sql.SparkSession;

public class RetainColumnGroupEx {

  static String warehouseLocation =System.getenv("SPARK_WAREHOUSE");
  static String dataHome = System.getenv("DATA_HOME")+"\\spark\\datasource\\csv";

  private static SparkSession start(){
    return SparkSession
      .builder()
      .master("local[*]")
      .appName("RollupDF")
      .config("spark.sql.warehouse.dir", warehouseLocation)
      .getOrCreate();
  }

  public static void main(String[] args) {
      try (SparkSession spark = start()) {
          System.out.println(spark.conf().get("spark.sql.retainGroupColumns"));
          System.out.println(spark.sessionState().conf().dataFrameRetainGroupColumns());
      }
  }

}

File: spark-scala/spark-dataframe/dataframe-aggregation/src/main/scala/com/mahfooz/datafram/aggregation/config/RetainColumnGroupEx.scala
package com.mahfooz.datafram.aggregation.config

import org.apache.spark.sql.SparkSession

object RetainColumnGroupEx {

  val warehouseLocation = sys.env.getOrElse("SPARK_WAREHOUSE","spark-warehouse")
  val dataHome = sys.env.getOrElse("DATA_HOME","data")+"\\spark\\datasource\\csv"

  private def start(): SparkSession = {
    SparkSession
      .builder()
      .master("local[*]")
      .appName("RollupDF")
      .config("spark.sql.warehouse.dir", warehouseLocation)
      .getOrCreate()
  }

  def main(args: Array[String]): Unit = {
    val spark=start()
    println(spark.conf.get("spark.sql.retainGroupColumns"))
    import spark.sessionState.conf
    println(conf.dataFrameRetainGroupColumns)
  }

}
